Ghana/GHA_transform_load_dlt.ipynb

```diff
@@ -0,0 +1,160 @@
+    "}\n",
+    "\n",
+    "@dlt.expect_or_drop(\"year_not_null\", \"YEAR IS NOT NULL\")\n",
+    "@dlt.table(name=f'gha_boost_bronze')\n",
+    "def boost_bronze():\n",
+    "    bronze_df = (spark.read\n",
+    "      .format(\"csv\")\n",
+    "      .options(**CSV_READ_OPTIONS)\n",
+    "      .option(\"inferSchema\", \"true\")\n",
+    "      .load(COUNTRY_MICRODATA_DIR)\n",
+    "    )\n",
+    "    for old_col_name in bronze_df.columns:\n",
+    "        new_col_name = old_col_name.replace(\" \", \"_\")\n",
+    "        bronze_df = bronze_df.withColumnRenamed(old_col_name, new_col_name)\n",
+    "    return bronze_df\n",
+    "\n",
+    "@dlt.table(name=f'gha_boost_silver')\n",
+    "def boost_silver():\n",
+    "    return (dlt.read(f'gha_boost_bronze')\n",
+    "            .withColumn('ECON1',coalesce(col('ECON1'), lit('')))\n",
+    "            .withColumn('FUnc1',coalesce(col('FUnc1'), lit('')))\n",
+    "            .withColumn('Func2',coalesce(col('Func2'), lit('')))\n",
+    "            .withColumn('Foreign',coalesce(col('Foreign'), lit('')))\n",
+    "            .filter(~(col('ECON1').startswith('32') | col('ECON1').startswith('33')))\n",
+    "            .withColumn('is_foreign', col('Foreign')!='n')\n",
+    "            .withColumn('admin0', lit('Central'))\n",
+    "            .withColumn('admin1', lit('Central Scope'))\n",
+    "            .withColumn('admin2', col('ADMIN1'))\n",
+    "            .withColumn('geo_tmp',\n",
+    "                when(col('GEO').substr(1, 2) == '00', 'Other')\n",
+    "                .when(col('GEO').substr(1, 2) == '01', 'Western')\n",
+    "                .when(col('GEO').substr(1, 2) == '02', 'Central')\n",
+    "                .when(col('GEO').substr(1, 2) == '03', 'Greater Accra')\n",
+    "                .when(col('GEO').substr(1, 2) == '04', 'Volta')\n",
+    "                .when(col('GEO').substr(1, 2) == '05', 'Eastern')\n",
+    "                .when(col('GEO').substr(1, 2) == '06', 'Ashanti')\n",
+    "                .when((col('GEO').substr(1, 2) == '07') & (col('YEAR')<2019), 'Brong Ahafo')\n",
+    "                .when((col('GEO').substr(1, 2) == '07') & (col('YEAR')>=2019), 'Bono')\n",
+    "                .when(col('GEO').substr(1, 2) == '08', 'Northern')\n",
+    "                .when(col('GEO').substr(1, 2) == '09', 'Upper East')\n",
+    "                .when(col('GEO').substr(1, 2) == '10', 'Upper West')\n",
+    "                .when(col('GEO').substr(1, 2) == '11', 'Oti')\n",
+    "                .when(col('GEO').substr(1, 2) == '12', 'Bono East')\n",
+    "                .when(col('GEO').substr(1, 2) == '13', 'Ahafo')\n",
+    "                .when(col('GEO').substr(1, 2) == '14', 'Savannah')\n",
+    "                .when(col('GEO').substr(1, 2) == '15', 'North East')\n",
+    "                .when(col('GEO').substr(1, 2) == '16', 'Western North')\n",
+    "                .when(col('GEO').substr(1, 2) == '99', 'Other')\n",
```

Comment:
Consider creating a new region_code column that takes on the value of `col('GEO').substr(1, 2)` before this step so you won't have to call `col('GEO').substr(1, 2)` so many times.